matchzoo/preprocessor/process_units.py

```diff
@@ -346,6 +347,47 @@ def transform(
         return hashing
 
 
+class SlidingWindowUnit(ProcessorUnit):
+    """
+    SlidingWindowUnit class.
+
+    Used to get information from a window sliding
+    through the input data.
+
+    Examples:
+        >>> data = np.array([[0,0,0],[1,1,1],[2,2,2],[3,3,3]])
+        >>> sliding = SlidingWindowUnit()
+        >>> output = sliding.transform(data)
+        >>> output.shape
+        (2, 9)
+        >>> output[0]
+        array([0, 0, 0, 1, 1, 1, 2, 2, 2])
+
+    """
+
+    def __init__(self, sliding_window: int=3):
+        """
+        Class initialization.
+
+        :param sliding_window: sliding window length.
+        """
+        self._sliding_window = sliding_window
+
+    def transform(self, inputs: np.ndarray) -> np.ndarray:
+        """
+        Compute the result of a window sliding through input data.
+
+        :param inputs: sequential input data.
+        :return: window sliding result.
+        """
+        output = []
+        while len(inputs) >= self._sliding_window:
+            output.append(np.concatenate(
+                inputs[:self._sliding_window], axis=-1))
+            inputs = inputs[1:]
+        return np.array(output)
```

Comment:
1. `return` should be inside `while` loop, right? if input length less than window size, it will return `np.array(input)`.
2. Do not forget to cover the `None` case in cdssm preprocessor.